sdk/eventhub/Microsoft.Azure.EventHubs/src/Primitives/EventHubsConnectionStringBuilder.cs

```diff
@@ -194,6 +195,11 @@ public EventHubsConnectionStringBuilder(string connectionString)
         /// </summary>
         public TransportType TransportType { get; set; }
 
+        /// <summary>
+        /// Enables Azure Active Directory Managed Identity authentication when set to 'Managed Identity'
```

Comment:
Thought about this option actually however enums don't really work well with connection string builder when default value needs to be NULL instead of enum value. Therefore, I decided to go with string literal for now. Full .Net SDK implementation is the same as this actually. We can revisit this once we receive feedback for new API.

Reply:
Maybe consider adding a pseudo-enumeration to allow discovery and help prevent typos, then?  Something like:

```c#
public static class AuthenticationType
{
    public static const ManagedIdentity = "Managed Identity";
}
```

That also gives you the advantage of being able to throw doc comments on the type and member to guide the user experience.

Reply:
Good idea.

---
In reply to: [289614634](https://github.com/Azure/azure-sdk-for-net/pull/6287#discussion_r289614634) [](ancestors = 289614634)

Reply:
Enums don't really work well with connection string builder. This has only 1 value for now. I will wait for some developer feedback on this.

---
In reply to: [287525933](https://github.com/Azure/azure-sdk-for-net/pull/6287#discussion_r287525933) [](ancestors = 287525933)